support-frontend/assets/components/paymentFrequencyButtons/paymentFrequencyButtons.tsx

```diff
@@ -0,0 +1,76 @@
+import type { SerializedStyles } from '@emotion/react';
+import { css } from '@emotion/react';
+import { from, palette, space, textSans } from '@guardian/source-foundations';
+import { useState } from 'react';
+
+interface PaymentFrequencyButtonObj {
+	label: string;
+	isPreSelected?: boolean;
+}
+
+interface PaymentFrequencyButtonsProps {
+	paymentFrequencies: PaymentFrequencyButtonObj[];
+	buttonClickHandler: (buttonIndex: number) => void;
+	additionalStyles?: SerializedStyles;
+}
+
+const container = (numberOfChildren: number) => css`
+	display: grid;
+	width: 100%;
+	grid-template-columns: repeat(${numberOfChildren}, 1fr);
+	gap: 1px;
+	${from.tablet} {
+		width: fit-content;
+	}
+`;
+
+const button = (isSelected: boolean) => css`
+	background-color: ${isSelected ? palette.neutral[100] : '#A2B2CB'};
+	transition: background-color 0.3s;
+	${textSans.medium({
+		fontWeight: 'bold',
+	})}
+	border: 0;
+	padding: ${space[3]}px ${space[9]}px;
+	color: ${palette.brand[400]};
+	:hover {
+		background-color: ${isSelected ? palette.neutral[100] : '#DAE0EA'};
+	}
+	cursor: pointer;
+	:first-child {
+		border-radius: ${space[3]}px 0 0 ${space[3]}px;
+	}
+	:last-child {
+		border-radius: 0 ${space[3]}px ${space[3]}px 0;
+	}
+`;
+
+export function PaymentFrequencyButtons({
+	paymentFrequencies,
+	buttonClickHandler,
+	additionalStyles,
+}: PaymentFrequencyButtonsProps): JSX.Element {
+	const [selectedButton, setSelectedButton] = useState(
+		Math.max(
```

Comment:
Is this needed now, can you have more than one selection of Monthly/Annual you need to get max for?

Reply:
the `findIndex` array method will return -1 if it doesn't find what are looking for. As we are using this to find an index  eg 0 or greater than 0 we use the Math.max method to ensure there is something to select.

Reply:
The scenario where -1 would be returned would be where none of the payment frequency objects in the `paymentFrequencies` array have the`isPreSelected` property set to true.